capabilityquerier/cmd/main.go

```diff
@@ -25,100 +23,131 @@ func failOnError(err error) {
 	}
 }
 
-func queryEndpoints(ctx context.Context,
-	listOfEndpoints *fetcher.ListOfEndpoints,
-	qw *capabilityquerier.QueueWorkers,
-	numWorkers int,
-	jobDuration time.Duration,
-	mq *lanternmq.MessageQueue,
-	ch *lanternmq.ChannelID,
-	qName string,
-	client *http.Client,
-	errs chan error,
-) {
-	err := qw.Start(ctx, numWorkers, errs)
-	failOnError(err)
+func queryEndpoints(message []byte, args *map[string]interface{}) error {
+	// Get arguments
+	qw, ok := (*args)["qw"].(*capabilityquerier.QueueWorkers)
+	if !ok {
+		return fmt.Errorf("unable to cast capabilityquerier QueueWorkers from arguments")
+	}
+	ctx, ok := (*args)["ctx"].(context.Context)
+	if !ok {
+		return fmt.Errorf("unable to cast context from arguments")
+	}
+	numWorkers, ok := (*args)["numWorkers"].(int)
+	if !ok {
+		return fmt.Errorf("unable to cast numWorkers to int from arguments")
+	}
+	errs, ok := (*args)["errs"].(chan error)
+	if !ok {
+		return fmt.Errorf("unable to cast errs to chan error from arguments")
+	}
+	client, ok := (*args)["client"].(*http.Client)
+	if !ok {
+		return fmt.Errorf("unable to cast client to http.Client from arguments")
+	}
+	jobDuration, ok := (*args)["jobDuration"].(time.Duration)
+	if !ok {
+		return fmt.Errorf("unable to cast jobDuration to time.Duration from arguments")
+	}
+	mq, ok := (*args)["mq"].(*lanternmq.MessageQueue)
+	if !ok {
+		return fmt.Errorf("unable to cast mq to MessageQueue from arguments")
+	}
+	ch, ok := (*args)["ch"].(*lanternmq.ChannelID)
+	if !ok {
+		return fmt.Errorf("unable to cast ch to ChannelID from arguments")
+	}
+	qName, ok := (*args)["qName"].(string)
+	if !ok {
+		return fmt.Errorf("unable to cast qName to string from arguments")
+	}
+
+	// Handle the start message that is sent before the endpoints and the stop message that is sent at the end
+	if string(message) == "start" {
+		err := qw.Start(ctx, numWorkers, errs)
+		failOnError(err)
+		return nil
+	}
+	if string(message) == "stop" {
+		log.Info("Stopping queue workers")
+		err := qw.Stop()
+		failOnError(err)
```

Comment:
why `failOnError` instead of returning an error or putting an error into the `errs` error channel?